### DISCUSSION TITLE
Incrementally building an array as a data document

### DISCUSSION THREAD

**Discussion by chamith-kodez** [open-policy-agent/community#413] [Community]
Hello Everyone,
Can someone please let me know how to add an element to an existing array which exists as a data document (probably using HTTP PATCH) ?

Existing data document:
[
    {
    "id": 7,
    "name": "resolution-ops-field-p1",
    "targetCode": "resolution",
    "scopeCode": "ops-field"
    },
    {
    "id": 8,
    "name": "resolution-client-p1",
    "targetCode": "resolution",
    "scopeCode": "client"
    }
]

**Comment by anderseknert** [comment]
Hi there! The OPA REST API documentation covers the PATCH operation [here](https://www.openpolicyagent.org/docs/latest/rest-api/#patch-a-document).

**Comment by chamith-kodez** [comment]
Thanks.
It does cover the Patch operation in a very basic manner. With that I cannot figure out how to keep adding new elements to an array.
________________________________

**Reply by anderseknert** [reply]
See the JSON Patch specification for that: https://jsonpatch.com/#operations

> Add
>
> { "op": "add", "path": "/biscuits/1", "value": { "name": "Ginger Nut" } }
>
> Adds a value to an object or inserts it into an array. In the case of an array, the value is inserted before the given index. The - character can be used instead of an index to insert at the end of an array.

**Reply by chamith-kodez** [reply]
Thanks @anderseknert a lot.
with this, I managed to get it done.
Looks like my knowledge on HTTP PATCH was minimal :)

**Reply by anderseknert** [reply]
I’m happy you got it working!
